components/eventForm/LocationBanner.tsx

```diff
@@ -41,49 +39,53 @@ export const EventFormLocationBanner = () => {
   )
 }
 
-const LocationInfoBanner = (locationInfo: EventFormLocationInfo) => {
-  return locationInfo.placemarkInfo
-    ? (
-      <PlacemarkInfoBanner {...locationInfo.placemarkInfo} />
-    )
-    : (
-      <GeocodedLocationInfoBanner {...locationInfo} />
-    )
-}
-
 const GeocodedLocationInfoBanner = (locationInfo: EventFormLocationInfo) => {
-  const { data: placemark, status } = useReverseGeocodeQuery(
-    locationInfo.coordinates
-  )
+  const { status, placemarkInfo } = usePlacemarkInfo(locationInfo)
 
   if (status === "error") {
     // TODO: - Should this just be the message of the error?
     return (
       <FormLabel
         style={styles.label}
-        icon="location-pin"
+        icon="location-on"
         headerText="Unable to find location, please try again later."
       />
     )
   }
 
   // TODO: - Add shimming loading effect
-  if (!placemark) return <SkeletonFormLabel icon="location-pin" />
+  if (!placemarkInfo) {
+    return <SkeletonFormLabel icon="location-on" />
+  }
 
-  const address = placemarkToFormattedAddress(placemark)
   return (
-    <PlacemarkInfoBanner name={placemark.name ?? undefined} address={address} />
+    <FormLabel
+      style={styles.label}
+      icon="location-on"
+      headerText={placemarkInfo.name ?? "Unknown Location"}
+      captionText={placemarkInfo.address ?? "Unknown Address"}
+    />
   )
 }
 
-const PlacemarkInfoBanner = ({ name, address }: EventFormPlacemarkInfo) => (
-  <FormLabel
-    style={styles.label}
-    icon="location-pin"
-    headerText={name ?? "Unknown Location"}
-    captionText={address ?? "Unknown Address"}
-  />
-)
+const usePlacemarkInfo = (locationInfo: EventFormLocationInfo) => {
+  const { data: placemark, status } = useReverseGeocodeQuery(
+    locationInfo.coordinates,
+    { enabled: !locationInfo.placemarkInfo }
```

Comment:
What state would this apply to?

Reply:
I basically forwarded the `enabled` option from react query, so it won't attempt to do any geocoding if there's already placemark info in the initial form values.